src/crd.rs

```diff
@@ -130,11 +130,34 @@ pub async fn wait_until_crds_present(
         //   3. queue another loop iteration if an error occurred and the timeout has not expired
         match check_result {
             Ok(()) => return Ok(()),
-            Err(err) => {
-                info!(
-                    "Error occurred when checking if all required CRDs are present: [{}]",
-                    err
+            Err(crate::error::Error::RequiredCrdsMissing { names }) => {
+                warn!(
+                    "The following required CRDs are missing from Kubernetes: [{:?}]",
+                    names
                 );
+                if let Some(timeout_value) = &timeout {
+                    if timeout.is_some() && start.elapsed() >= *timeout_value {
+                        info!(
+                            "Timeout of [{}] seconds reached, returning.",
+                            timeout_value.as_secs()
+                        );
+                        return Err(RequiredCrdsMissing { names });
+                    }
+                }
+            }
```

Comment:
Isn't this a duplicate now?
You handle this case below already, am I missing something?

Reply:
Nope, you are completely right. That was an earlier attempt at addressing your comment which I forgot to clean up.